components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt

```diff
@@ -77,17 +81,18 @@ internal val sharedDiskCache = IconDiskCache()
  */
 class BrowserIcons @Suppress("LongParameterList") constructor(
     private val context: Context,
-    private val httpClient: Client,
+    httpClient: Client,
+    deferLoadingNetworkIcons: Boolean = false,
```

Comment:
I think you mentioned adding a flag to the request object. What made you move it to the class? Since `BrowserIcons` is a singleton I am expecting that not all code will want to defer loads? 🤔

Reply:
Looked into this but saw the `IconRequest` being instantiated in suggestion providers or various other parts of the app and containing only the info for what icons to load. 
I'd say that what happens after with those icons, when and how they are used shouldn't be controlled from these places where the `IconRequest` is created, that being another functionality that these places shouldn't know or care about. Having to set the right "don't wait for network" flag in this many places also seems bound to cause issues (strange mixes) or just be forgotten about.
And even if we'd had that flag in the request we'd still need all the changes in this PR (minus the flag in `BrowserIcons`) to actually have a different behavior based on that.
So it made a bit more sense to me to have this flag above the requests, in one single place, where the entire functionality - for what & how to download all the way to what to return is set up.

If different app features - clients of the icons functionality want a different behavior like wait / not wait for network icons they should have an easy api for this and I think the current constructor allows for this, though if clients want a different behavior they should use different instances of `BrowserIcons`.

Reply:
I think I disagree here. I believe in most cases it is okay to wait for the load. That's why this is returning a deferred Icon. Most places can display a placeholder or nothing and display an icon once it's available. (which eventually should be the behavior of the AwesomeBar too -> #11185). There's no need to not display an icon after the network call for all asynchronous loads. I also feel it is way less risky to keep the existing behavior and change the request in places where we are sure this will have an impact.

> though if clients want a different behavior they should use different instances of BrowserIcons.

That's not how it is designed though. It is meant to be a singleton.

Reply:
Refactored now the patch to have this flag in the `IconRequest.` based on which the same instance of `BrowserIcons` would block on downloading icons or do this in background.